autogpt_platform/frontend/src/app/(platform)/marketplace/agent/[creator]/[slug]/page.tsx

```diff
@@ -1,124 +1,74 @@
-import BackendAPI from "@/lib/autogpt-server-api";
-import { Breadcrumbs } from "@/components/molecules/Breadcrumbs/Breadcrumbs";
-import { AgentInfo } from "@/components/agptui/AgentInfo";
-import { AgentImages } from "@/components/agptui/AgentImages";
-import { AgentsSection } from "@/components/agptui/composite/AgentsSection";
-import { BecomeACreator } from "@/components/agptui/BecomeACreator";
-import { Separator } from "@/components/ui/separator";
 import { Metadata } from "next";
 import { getServerUser } from "@/lib/supabase/server/getServerUser";
+import {
+  getV2GetSpecificAgent,
+  prefetchGetV2GetSpecificAgentQuery,
+  prefetchGetV2ListStoreAgentsQuery,
+} from "@/app/api/__generated__/endpoints/store/store";
+import { StoreAgentDetails } from "@/app/api/__generated__/models/storeAgentDetails";
+import { MainAgentPage } from "../../../components/MainAgentPage/MainAgentPage";
+import { getQueryClient } from "@/lib/react-query/queryClient";
+import { prefetchGetV2GetAgentByStoreIdQuery } from "@/app/api/__generated__/endpoints/library/library";
+import { dehydrate, HydrationBoundary } from "@tanstack/react-query";
 
-// Force dynamic rendering to avoid static generation issues with cookies
 export const dynamic = "force-dynamic";
 
-type MarketplaceAgentPageParams = { creator: string; slug: string };
+export type MarketplaceAgentPageParams = { creator: string; slug: string };
 
 export async function generateMetadata({
   params: _params,
 }: {
   params: Promise<MarketplaceAgentPageParams>;
 }): Promise<Metadata> {
-  const api = new BackendAPI();
   const params = await _params;
-  const agent = await api.getStoreAgent(params.creator, params.slug);
-
+  const { data: creator_agent } = await getV2GetSpecificAgent(
+    params.creator,
+    params.slug,
+  );
   return {
-    title: `${agent.agent_name} - AutoGPT Marketplace`,
-    description: agent.description,
+    title: `${(creator_agent as StoreAgentDetails).agent_name} - AutoGPT Marketplace`,
+    description: (creator_agent as StoreAgentDetails).description,
```

Comment:
If data is loading or the request fails I think we would have a bad error here. Let's be more defensive an do something like:
```ts
const { data: creatorAgent, isSuccess } = await getV2GetSpecificAgent(
   params.creator,
   params.slug,
);
```